src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return (List<T>) new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return (List<T>) new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            return (List<T>) new TruckProducer().get();
+        }
```

Comment:
Keep if-else constructions simple. Since you are returning a value inside each if clause, you can omit the `else` keyword to reduce nesting and improve readability. This is known as the guard clause pattern.